daemon/src/oracle.rs

```diff
@@ -192,6 +204,13 @@ impl<CFD: 'static, M: 'static> xtra::Handler<GetAnnouncement> for Actor<CFD, M>
     }
 }
 
+#[async_trait]
+impl<CFD: 'static, M: 'static> xtra::Handler<NewAnnouncementFetched> for Actor<CFD, M> {
+    async fn handle(&mut self, msg: NewAnnouncementFetched, _ctx: &mut xtra::Context<Self>) {
+        self.latest_announcements.insert(msg.id, msg.announcement);
```

Comment:
:question: I know this works, but why do we need to send a message to ourselves instead of just directly inserting the new announcements after completing the for loop in `update_latest_announcements`? This seems a bit too indirect.

Reply:
Okay, that makes sense. I didn't realise we were spawning tasks to fetch the announcements in parallel, which is what the title of the commit says :facepalm: